app/controllers/VrtController.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import javax.inject.Inject
+import model.{PeriodKey, VrtId, VrtRepaymentDetailData}
+import play.api.Logger
+import play.api.mvc.{ControllerComponents, Request}
+import repository.VrtRepo
+import uk.gov.hmrc.play.bootstrap.controller.BackendController
+
+import scala.concurrent.ExecutionContext
+
+abstract class VrtController @Inject() (cc: ControllerComponents, repo: VrtRepo)
+  (implicit executionContext: ExecutionContext) extends BackendController(cc) {
+  private[controllers] def store()(implicit request: Request[VrtRepaymentDetailData]) = {
```

Comment:
We rather prefer injection of functionality via some service/util instead of using inheritance. 
If it's not a big problem, could you move the `def store(...)` method into some service and inject this service in `Controller` and `TestController` please?